src/ui-input.c

```diff
@@ -976,26 +1020,57 @@ static bool get_file_text(const char *suggested_name, char *path, size_t len)
 
 	/* Get filename */
 	my_strcpy(buf, suggested_name, sizeof buf);
-	if (!get_string("File name: ", buf, sizeof buf)) return false;
 
-	/* Make sure it's actually a filename */
-	if (buf[0] == '\0' || buf[0] == ' ') return false;
+    if(!arg_force_name) {
 
-	/* Build the path */
-	path_build(path, len, ANGBAND_DIR_USER, buf);
+        if (!get_string("File name: ", buf, sizeof buf)) return false;
 
-	/* Check if it already exists */
-	if (file_exists(path) && !get_check("Replace existing file? "))
-		return false;
+        /* Make sure it's actually a filename */
+        if (buf[0] == '\0' || buf[0] == ' ') return false;
 
-	/* Tell the user where it's saved to. */
-	prt(format("Saving as %s.", path), 0, 0);
-	anykey();
-	prt("", 0, 0);
+    }
 
-	return true;
+    /* Build a filename with a timestamp for users who are playing on a server */
+    else {
+
+        int len;
+        time_t ltime;
+        struct tm *today;
+        
+        /* Get the current time */
+        time(&ltime);
+        today = localtime(&ltime);
+
+        prt("File name: ", 0,0);
+
+        /* Overwrite the ".txt" that was added */
+        len = strlen(buf)-4;
+        strftime(buf+len, sizeof(buf)-len, "_%Y%b%d_%H%M%S.txt", today);
+    
+        /* Prompt the user to confirm or cancel the file dump */
+        if (!askfor_aux(buf, sizeof buf, get_filename_keypress))
```

Comment:
Instead of using askfor_aux here, since all the user can do is say yes or no, I'd suggest using get_check() instead to ask "File will be saved as %s. Is this OK? " or similar.